rust-runtime/aws-smithy-http-client/src/lib.rs

```diff
@@ -20,24 +21,61 @@
     unreachable_pub,
     rust_2018_idioms
 )]
+#![cfg_attr(docsrs, feature(doc_cfg))]
 
 // ideally hyper_014 would just be exposed as is but due to
 // https://github.com/rust-lang/rust/issues/47238 we get clippy warnings we can't suppress
 #[cfg(feature = "hyper-014")]
 pub(crate) mod hyper_legacy;
 
-/// Default HTTP and TLS connectors that use hyper 0.14.x and rustls.
+/// Legacy HTTP and TLS connectors that use hyper 0.14.x and rustls.
 #[cfg(feature = "hyper-014")]
 #[deprecated = "hyper 0.14.x support is deprecated, please migrate to 1.x client"]
 pub mod hyper_014 {
     pub use crate::hyper_legacy::*;
 }
 
-// TODO(https://github.com/smithy-lang/smithy-rs/issues/1925) - do we even want to name this/tie this to hyper?
-
-/// Default HTTP and TLS connectors that use hyper 0.14.x and rustls.
+/// Default HTTP and TLS connectors
+#[cfg(feature = "hyper-1")]
+pub(crate) mod client;
 #[cfg(feature = "hyper-1")]
-pub mod hyper_1;
+pub use client::{default_client, tls, Builder, Connector, ConnectorBuilder};
 
 #[cfg(feature = "test-util")]
 pub mod test_util;
+
+#[allow(unused_macros, unused_imports)]
+#[macro_use]
+pub(crate) mod cfg {
+    /// Any TLS provider enabled
+    macro_rules! cfg_tls {
```

Comment:
Is this intentionally the same as `cfg_rustls!`?

Reply:
Right now it is yes. `cfg_tls` is meant for _any_ TLS provider enabled (with the expectation we'll be adding at least opt-in support for S2n in the near future) whereas `cfg_rustls` is targeted to if rustls is enabled.